src/headless/converse-emoji.js

```diff
@@ -147,7 +147,7 @@ export function getEmojiMarkup (data, options={unicode_only: false, add_title_wr
 export function getShortnameReferences (text) {
     const references = [...text.matchAll(shortnames_regex)];
     return references.map(ref => {
-        const cp = converse.emojis.by_sn[ref[0]].cp;
+        const cp = converse.emojis.by_sn?.[ref[0]]?.cp;
         return {
             cp,
             'begin': ref.index,
```

Comment:
Thanks for your feedback, I agree with the optional chaining issues, and will attempt to be as explicit as possible on the next MR.

I will add some info to the description now.

Reply:
That being said, I think if you have to know about all the things that have to be waited for before calling certain functions, it will be easy to fall in these traps and come across bugs such as this. I wish there was a better solution for these async tasks.

Reply:
Yeah, that's why I now raise an error in that method and tell the user (i.e. developer) to wait for the promise.

Reply:
Thanks for updating.